app/lib/actions/katello/agent_action.rb

```diff
@@ -0,0 +1,106 @@
+module Actions
+  module Katello
+    class AgentAction < Actions::EntryAction
+      include Dynflow::Action::Timeouts
+      include Helpers::Presenter
+
+      def self.agent_message
+        fail NotImplementedError
+      end
+
+      def agent_action_type
+        nil
+      end
+
+      def plan(host, options)
+        action_subject(host)
+
+        # if already dispatched by bulk action use the provided history ID
+        dispatch_history_id = options[:dispatch_histories]&.dig(host.id.to_s)
```

Comment:
will this do the same thing ?

```suggestion
        dispatch_history_id = options.dig(:dispatch_histories, host.id.to_s)
```

Reply:
Yes!